zebra-chain/src/sprout/tree.rs

```diff
@@ -116,186 +121,180 @@ impl From<&Root> for [u8; 32] {
     }
 }
 
-/// Sprout Note Commitment Tree
-#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
-#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]
+/// A node of the Sprout note commitment tree.
+#[derive(Clone, Debug)]
+struct Node([u8; 32]);
+
+impl incrementalmerkletree::Hashable for Node {
+    fn empty_leaf() -> Self {
+        Self(NoteCommitmentTree::uncommitted())
+    }
+
+    /// Combines two nodes to generate a new node.
+    ///
+    /// Note that Sprout does not use the `level` argument.
+    ///
+    /// https://zips.z.cash/protocol/protocol.pdf#sproutmerklecrh
+    fn combine(_level: incrementalmerkletree::Altitude, a: &Self, b: &Self) -> Self {
+        Self(merkle_crh_sprout(a.0, b.0))
+    }
+
+    /// Returns the node for the level below the given level. (A quirk of the API)
+    fn empty_root(level: incrementalmerkletree::Altitude) -> Self {
+        let layer_below: usize = MERKLE_DEPTH - usize::from(level);
```

Comment:
(See other comment)

Or don't do this conversion. but still get the level above/below, it's confusing. (I think I named one of these wrong, because if it's the level below then it's the layer above)